src/table/sticky-columns/utils.ts

```diff
@@ -0,0 +1,51 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import { CellOffsets, StickyColumnsCellState, StickyColumnsProps, StickyColumnsWrapperState } from './interfaces';
+
+export function isCellStatesEqual(s1: null | StickyColumnsCellState, s2: null | StickyColumnsCellState): boolean {
+  if (s1 && s2) {
+    return (
+      s1.padLeft === s2.padLeft &&
+      s1.lastLeft === s2.lastLeft &&
+      s1.lastRight === s2.lastRight &&
+      s1.offset.left === s2.offset.left &&
+      s1.offset.right === s2.offset.right
+    );
+  }
+  return s1 === s2;
+}
+
+export function isWrapperStatesEqual(s1: StickyColumnsWrapperState, s2: StickyColumnsWrapperState): boolean {
+  return s1.scrollPaddingLeft === s2.scrollPaddingLeft && s1.scrollPaddingRight === s2.scrollPaddingRight;
+}
+
+export function updateCellOffsets(cells: Record<PropertyKey, HTMLElement>, props: StickyColumnsProps): CellOffsets {
+  const firstColumnsWidths: number[] = [];
+  for (let i = 0; i < Math.min(props.visibleColumns.length, props.stickyColumnsFirst); i++) {
+    const element = cells[props.visibleColumns[i]];
+    const cellWidth = element.getBoundingClientRect().width ?? 0;
+    firstColumnsWidths[i] = (firstColumnsWidths[i - 1] ?? 0) + cellWidth;
+  }
+
+  const lastColumnsWidths: number[] = [];
+  for (let i = Math.min(props.visibleColumns.length, props.stickyColumnsLast) - 1; i >= 0; i--) {
+    const element = cells[props.visibleColumns[i]];
+    const cellWidth = element.getBoundingClientRect().width ?? 0;
+    lastColumnsWidths[i] = (lastColumnsWidths[i + 1] ?? 0) + cellWidth;
+  }
+  lastColumnsWidths.reverse();
+
+  const stickyWidthLeft = firstColumnsWidths[props.stickyColumnsFirst - 1] ?? 0;
+  const stickyWidthRight = lastColumnsWidths[props.stickyColumnsLast - 1] ?? 0;
+  const offsets = props.visibleColumns.reduce(
+    (map, columnId, columnIndex) =>
+      map.set(columnId, {
+        first: firstColumnsWidths[columnIndex - 1] ?? 0,
+        last: lastColumnsWidths[props.visibleColumns.length - 1 - columnIndex - 1] ?? 0,
```

Comment:
So for non sticky columns there will be `0` value. I guess, it currently works fine, because this value is never used, but it is definitely a confusion when debugging the code and seeing these phantom values

Reply:
Agree, it would be nice to choose a better data structure going forward. The idea is that we don't need to iterate over all columns when only sticky columns are needed.